modules/rte_mis_state/rte_mis_state.module

```diff
@@ -293,3 +296,35 @@ function rte_mis_state_form_user_form_alter(&$form, &$form_state, $form_id) {
   }
 
 }
+
+/**
+ * Implements hook_views_query_alter().
+ */
+function rte_mis_state_views_query_alter(ViewExecutable $view, QueryPluginBase $query) {
+  if ($view->id() == 'school_udise_code' && $view->current_display == 'page_1') {
+    $user = \Drupal::currentUser();
+    $curr_user = User::load($user->id());
+    $curr_user_role = $curr_user->getRoles();
+    if (in_array('district_admin', $curr_user_role)) {
+      $definition = [
+        'table' => 'taxonomy_term__field_location',
+        'field' => 'entity_id',
+        'left_table' => 'taxonomy_term_field_data',
+        'left_field' => 'tid',
+        'type' => 'INNER',
+      ];
+      $join = Drupal::service('plugin.manager.views.join')->createInstance('standard', $definition);
+      $query->addRelationship('taxonomy_term__field_location', $join, 'taxonomy_term_field_data');
+      $curr_user_location = $curr_user->get('field_location_details')->getValue()[0]['target_id'];
```

Comment:
Use `->getString()` function to get the target id

Reply:
Ok.